dstep/translator/Record.d

```diff
@@ -59,9 +61,17 @@ private:
 
     string writeRecord (string name, void delegate (Data context) dg)
     {
+        auto known = cursor.definition in this.record_definitions;
+        if (known)
+            return "";
+
         auto context = new Data;
-        static if (is(typeof(context.isFwdDeclaration) == bool))
-            context.isFwdDeclaration = !cursor.isDefinition;
+        
+        if (cursor.isDefinition)
+            this.record_definitions[cursor] = true;
+        else
+            context.isFwdDeclaration = true;
+
```

Comment:
Ok, thanks, I have missed step 6 where it went in recursion and was confused by seemingly broken iteration over struct members.